pkg/dispatchcli/cmd/get_run.go

```diff
@@ -58,35 +81,130 @@ func getFunctionRun(out, errOut io.Writer, cmd *cobra.Command, args []string, c
 	fnName := args[0]
 	runName := args[1]
 
-	resp, err := client.GetFunctionRun(context.TODO(), "", fnName, runName)
+	since := time.Now()
+	resp, err := client.GetFunctionRun(context.TODO(), "", fnName, runName, nil)
 
 	if err != nil {
 		return err
 	}
-	return formatRunOutput(out, false, []v1.Run{*resp})
+	if err = formatRunOutput(out, false, true, []v1.Run{*resp}); err != nil {
+		return err
+	}
+	if followRuns {
+		if err = followFilteredRuns(out, c, &fnName, &runName, since); err != nil {
+			return err
+		}
+	}
+	return nil
 }
 
 func getRuns(out, errOut io.Writer, cmd *cobra.Command, args []string, c client.FunctionsClient) error {
-	resp, err := c.ListRuns(context.TODO(), "")
+	since := time.Now()
+	resp, err := c.ListRuns(context.TODO(), "", nil)
 
 	if err != nil {
 		return err
 	}
-	return formatRunOutput(out, true, resp)
+	if last && len(resp) > 0 {
+		lastRun := resp[0]
+		for _, run := range resp {
```

Comment:
This is ok for now, but it seems like something that should be done server-side, i.e. sort by execution time and then retrieve the first result. I would imagine the number of runs for single will grow considerably, and some pagination/filtering on the server side will need to be added anyway (like you did with "since")

Reply:
Yeah the number of runs could definitely get considerably large so this is more of a temp solution

Reply:
+1